config/configmapprovider/file.go

```diff
@@ -47,25 +47,23 @@ func NewFile() Provider {
 	return &fileMapProvider{}
 }
 
-func (fmp *fileMapProvider) Retrieve(_ context.Context, location string, _ WatcherFunc) (Retrieved, error) {
+func (fmp *fileMapProvider) Retrieve(_ context.Context, location string, _ WatcherFunc) (*config.Map, EndWatchFunc, error) {
```

Comment:
Given this is an API, would it be better to return a specific type containing the map and the func? A func that returns 3 values leaves a bad aftertaste, especially for a pre-v1 API, in my opinion.

Reply:
So you think the alternative consideration is better. I think having a simple struct with 2 public members is as good as this and definitely better than an interface (as it was before).

Reply:
My reasoning is that it's easy to add a new field to the return object type if needed, without breaking current callers, while if you add an extra value to the return values, you break them. For some reason, I think it's easier to start needing 4 values when you have 3 than needing 3 when you have 2 (one of them being `error`).

Reply:
Sounds reasonable to me, will either change this or maybe do another one to look at that option as well.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/pull/4789 here is the alternative, let me know.